File: my-submissions/e744.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int left = 0;
        int right = letters.length - 1;


        while (left < right) {
            int mid = (left + right) / 2;
            
            if (letters[mid] <= target) {
                left = mid + 1;
                continue;
            }

            right = mid - 1;
        }
        
        if (letters[left] <= target)
            return letters[(left + 1) % letters.length];
        return letters[left];
    }
}

File: my-submissions/e744.py
class Solution:
    def nextGreatestLetter(self, letters: List[str], target: str) -> str:
        left, right = 0, len(letters) - 1

        while left < right :
            mid = (left + right) // 2
            
            if ord(letters[mid]) <= ord(target) :
                left = mid + 1
                continue

            right = mid - 1
        
        if (ord(letters[left]) <= ord(target)) :
            return letters[(left + 1) % len(letters)]
        return letters[left]